include/civetweb.h

```diff
@@ -334,6 +334,15 @@ mg_set_websocket_handler(struct mg_context *ctx,
                          mg_websocket_close_handler close_handler,
                          void *cbdata);
 
+/* mg_set_auth_handler
+
+   Sets or removes a URI mapping for an authorization handler.
+   This function works similar to mg_set_request_handler - see there. */
+CIVETWEB_API void mg_set_auth_handler(struct mg_context *ctx,
+                                      const char *uri,
+                                      mg_request_handler handler,
```

Comment:
Agreed. Maybe the best way to clean this up is to restore 'mg_set_request_handler_type' to it's original state and just implement adding the mg_authorization_handler directly in 'mg_set_auth_handler' and not try to reuse code in  'mg_set_request_handler_type'.